Python-packages/covidcast-py/covidcast/plotting.py

```diff
@@ -1,18 +1,55 @@
 """This contains the plotting and geo data management methods for the COVIDcast signals."""
 
 import os
+from datetime import date
+from typing import Tuple
 
 import geopandas as gpd
+import numpy as np
 import pandas as pd
-
 from covidcast import covidcast
+from matplotlib import pyplot as plt
 
 SHAPEFILE_PATHS = {"county": "../shapefiles/county/cb_2019_us_county_5m.shp",
                    "state": "../shapefiles/state/cb_2019_us_state_5m.shp"}
 
 CURRENT_PATH = os.path.dirname(os.path.realpath(__file__))
 
 
+def plot_choropleth(data: pd.DataFrame,
+                    time_value: date = None,
+                    **kwargs) -> gpd.GeoDataFrame:
+    """Given the output df of covidcast.signal(), plot the choropleth.
+
+    Plot is in a style similar to the website https://covidcast.cmu.edu.
+
+    :param data: DataFrame of values and geographies
+    :param time_value: date of interest for the geography
+    :param kwargs: optional keyword arguments passed to plot()
+    :return: GeoDF with polygon info. Same as get_geo_df()
+    """
+    data_source, signal, geo_type = covidcast._detect_metadata(data)  # pylint: disable=W0212
+    meta = covidcast._signal_metadata(data_source, signal, geo_type)  # pylint: disable=W0212
+    # use most recent date in data if none provided
+    day_to_plot = time_value if time_value else max(data.time_value)
+    day_data = data.loc[data.time_value == day_to_plot, :]
+    data_w_geo = get_geo_df(day_data)
+    _, ax = plt.subplots(1, figsize=(40, 12))
+    ax.axis("off")
+
+    kwargs["vmin"] = kwargs.get("vmin", 0)
+    kwargs["vmax"] = kwargs.get("vmax", meta["mean_value"] + 3 * meta["stdev_value"])
+    kwargs["cmap"] = kwargs.get("cmap", "YlOrRd")
+
+    sm = plt.cm.ScalarMappable(cmap=kwargs["cmap"])
+    plt.colorbar(sm, boundaries=np.linspace(kwargs["vmin"], kwargs["vmax"], 8),
+                 orientation="horizontal", fraction=0.02, pad=0.05)
+    plt.title(f"{data_source}: {signal}, {day_to_plot.strftime('%Y-%m-%d')}")
+    for shape in _project_and_transform(data_w_geo):
+        shape.plot("value", ax=ax, **kwargs)
+    return data_w_geo
```

Comment:
It's been a long time since I used matplotlib. But should we return a matplotlib object here, such as the figure object, so the user can further customize the plot by calling various matplotlib functions?

Reply:
I'm not very familiar with mpl practices but that sounds right. Will change to return the figure.